### PR TITLE
[Network] az network vhub route reset: reset this route when the routingState is set to Failed.

### PR DESCRIPTION
**Pull request by MyronFanQiu**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

Fix Azure/azure-cli#13363

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Network

File: src/virtual-wan/azext_vwan/_help.py [hunk 1]
```diff
@@ -87,6 +87,11 @@
     short-summary: Remove a route from the virtual hub route table.
 """
 
+helps['network vhub route reset'] = """
+    type: command
+    short-summary: Reset virtual hub route when the route state is set to failed.
```
Comment (jsntcy) [inline src/virtual-wan/azext_vwan/_help.py:92]: is failed

File: src/virtual-wan/azext_vwan/custom.py [hunk 2]
```diff
@@ -196,6 +196,21 @@ def list_hub_routes(cmd, resource_group_name, virtual_hub_name):
...
+    client = network_client_factory(cmd.cli_ctx).virtual_hubs
+    hub = client.get(resource_group_name, virtual_hub_name)
+    if hub.routing_state == 'Failed':
+        logger.warning('Reset virtual hub')
+        poller = sdk_no_wait(no_wait, client.create_or_update,
+                             resource_group_name, virtual_hub_name, hub)
+        try:
+            return poller.result().route_table.routes
+        except AttributeError:
+            return
```
Comment (jsntcy) [inline src/virtual-wan/azext_vwan/custom.py:209]: >return [](start = 12, length = 6)

routing state

File: src/virtual-wan/azext_vwan/custom.py [hunk 3]
```diff
@@ -196,6 +196,21 @@ def list_hub_routes(cmd, resource_group_name, virtual_hub_name):
...
+    logger.warning("Virtual Hub's state is not `failed`. Skip this command")
+    return
```
Comment (jsntcy) [inline src/virtual-wan/azext_vwan/custom.py:211]: remove "return"?

**Pull request merged**